pkg/gatewayserver/io/ws/ws.go

```diff
@@ -249,23 +211,29 @@ func (s *srv) handleTraffic(c echo.Context) (err error) {
 		return err
 	}
 
+	// Handle session state
+	err = s.formatter.Connect(ctx, uid)
+	if err != nil {
+		logger.WithError(err).Warn("Failed to handle session")
+		return err
+	}
```

Comment:
So here basically `WithSession()`, deriving a context with state that can be consumed by the formatter